kythe/typescript/README.md

```diff
@@ -1,5 +1,269 @@
+members, which belong to the class value, have a form of `$CLASS.$PROPERTY`.
+
+**SyntaxKind**:
+
+-   PropertyDeclaration
+-   ParameterPropertyDeclaration
+-   GetAccessor
+-   SetAccessor
+-   MethodDeclaration
+
+```typescript
+class Klass {
+  //- @prop defines/binding VName("Klass.prop", _, _, _, _)
+  prop;
+
+  //- @prop defines/binding VName("Klass#type.prop", _, _, _, _)
+  static prop;
+
+  //- @cprop defines/binding VName("Klass#type.cprop", _, _, _, _)
+  constructor(private cprop: number) {}
+}
+```
+
+#### Getter/Setter
+
+**Form**:
+
+-   getter: `$DECLARATION_NAME:getter`
+-   setter: `$DECLARATION_NAME:setter`
+-   getter if present, otherwise setter: `$DECLARATION_NAME`
+
+**Notes**: Because getters and setters refer to the same property on a class but
+provide multiple declarations, several distinctions between the declarations are
+made. Each declaration of a getter and setter will take `:getter` and `:setter`
+suffix, respectively.
+
+If a getter for a property is present, the getter will emit a binding the same
+form as the property it defines. If only a setter is present, the setter will
+emit a binding of the same form as the property it defines.
+
+**SyntaxKind**:
+
+-   GetAccessor
+-   SetAccessor
+
+```typescript
+class Klass {
+  //- @foo defines/binding VName("Klass.foo:getter", _, _, _, _)
+  //- @foo defines/binding VName("Klass.foo", _, _, _, _)
+  get foo() {}
+  //- @foo defines/binding VName("Klass.foo:setter", _, _, _, _)
+  set foo(newFoo) {}
+}
+
+class KlassNoGetter {
+  //- @foo defines/binding VName("KlassNoGetter.foo:setter", _, _, _, _)
+  //- @foo defines/binding VName("KlassNoGetter.foo", _, _, _, _)
+  set foo(newFoo) {}
+}
+```
+
+#### Export Assignment
+
+**Form**: `default`
+
+**Notes**: `export default` is semantically equivalent to exporting a variable
+named `default`. `export =` is currently not supported.
+
+**SyntaxKind**:
+
+-   `ExportAssignment`
+
+```typescript
+//- @default defines/binding VName("default", _, _, _, _)
+export default myExport;
+```
+
+#### Anonymous Block
+
+**Form**: a unique, non-deterministic block name.
+
+**Notes**: The block name is not guaranteed, because declarations within an
+anonymous block cannot be accessed outside it.
+
+**SyntaxKind**:
+
+-   `ArrowFunction`
+-   `Block` that does not have a `FunctionDeclaration` or `MethodDeclaration`
+    parent
+
+```typescript
+let af = () => {
+  //- @decl defines/binding VName(_, _, _, _, _)
+  let decl;
+};
+```
+
+### VName corpus
+
+Project-specific, defined by the compilation unit you pass to the indexer.
+
+### VName root
+
+Project-specific, defined by the compilation unit you pass to the indexer.
+
+### VName path
+
+-   For entire source code files
+    -   the entire file path, relative to the corpus and root.
+-   For declarations with a file
+    -   the file path stripped of `.d.ts` or `.ts` extensions, relative to the
+        corpus and root.
```

Comment:
This might be better offloading to the module name discussion already in the readme.

Reply:
I agree that is less brittle, but it also makes reading the specification much harder. I'll try it out though.